File: 1 - Two Sum.py
from collections import defaultdict
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        numsDict = defaultdict(list)
        for index, element in enumerate(nums):
            numsDict[element].append(index)
        for i in numsDict:
            if target-i in numsDict:
                if target-i != i:
                    return [numsDict[target-i][0],numsDict[i][0]]
                elif len(numsDict[target-i]) > 1:
                    return numsDict[target-i]

# simple & elegant
class Solution(object):
    def twoSum(self, nums, target):
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if nums[i]+nums[j]==target:
                    return [i,j]

File: 1 - Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans(2);
        for(int i = 0; i < nums.size() - 1; i++) {
            for(int j = i + 1; j < nums.size(); j++) {
                if(nums[i] + nums[j] == target) {
                    ans[0] = i;
                    ans[1] = j;
                    return ans;
                }
            }
        }
        
    }
};
